x-pack/plugins/apm/public/components/fleet_integration/apm_policy_form/index.tsx

```diff
@@ -5,30 +5,128 @@
  * 2.0.
  */
 import { EuiSpacer } from '@elastic/eui';
-import React from 'react';
-import { OnFormChangeFn, PackagePolicyVars } from './typings';
-import { APMSettingsForm } from './settings/apm_settings';
-import { RUMSettingsForm } from './settings/rum_settings';
-import { TLSSettingsForm } from './settings/tls_settings';
+import { i18n } from '@kbn/i18n';
+import React, { useMemo } from 'react';
+import {
+  getAgentAuthorizationSettings,
+  isAgentAuthorizationFormValid,
+} from './settings_definition/agent_authorization_settings';
+import {
+  getApmSettings,
+  isAPMFormValid,
+} from './settings_definition/apm_settings';
+import {
+  getRUMSettings,
+  isRUMFormValid,
+} from './settings_definition/rum_settings';
+import {
+  getTLSSettings,
+  isTLSFormValid,
+} from './settings_definition/tls_settings';
+import { SettingsForm } from './settings_form';
+import { mergeNewVars } from './settings_form/utils';
+import { PackagePolicyVars } from './typings';
 
 interface Props {
-  onChange: OnFormChangeFn;
+  updateAPMPolicy: (newVars: PackagePolicyVars, isValid: boolean) => void;
   vars?: PackagePolicyVars;
   isCloudPolicy: boolean;
 }
 
-export function APMPolicyForm({ vars = {}, isCloudPolicy, onChange }: Props) {
+export function APMPolicyForm({
+  vars = {},
+  isCloudPolicy,
+  updateAPMPolicy,
+}: Props) {
+  const {
+    apmSettings,
+    rumSettings,
+    tlsSettings,
+    agentAuthorizationSettings,
+  } = useMemo(() => {
+    return {
+      apmSettings: getApmSettings(isCloudPolicy),
+      rumSettings: getRUMSettings(),
+      tlsSettings: getTLSSettings(),
+      agentAuthorizationSettings: getAgentAuthorizationSettings(isCloudPolicy),
+    };
+  }, [isCloudPolicy]);
+
+  function handleFormChange(key: string, value: any) {
+    // Merge new key/value with the rest of fields
+    const newVars = mergeNewVars(vars, key, value);
+
+    // Validate the entire form before sending it to fleet
+    const isFormValid =
+      isAPMFormValid(newVars, apmSettings) &&
+      isRUMFormValid(newVars, rumSettings) &&
+      isTLSFormValid(newVars, tlsSettings) &&
+      isAgentAuthorizationFormValid(newVars, agentAuthorizationSettings);
+
+    updateAPMPolicy(newVars, isFormValid);
+  }
+
+  const settingsDefinition = [
```

Comment:
What about calling it a `Section` or `SettingsSection` and then adding a type:

```suggestion
  const settingsSection: SettingsSection = [
```

Then you can use `SettingsSection` in `SettingsForm`